traffic_portal/test/end_to_end/serverCapabilities/pageData.js

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+module.exports = function(){
+	this.name=element(by.name('name'));
+	this.createButton=element(by.buttonText('Create'));
+	this.deleteButton=element(by.buttonText('Delete'));
+	this.searchFilter=element(by.id('serverCapabilitiesTable_filter')).element(by.css('label')).element(by.css('input'));
```

Comment:
can't you combine this into one `by.css`? `by.css('label input')`?

Reply:
maybe but they seem fine separate to me. i'm sure all of this UI test framework could be cleaned up.